modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,194 @@
+ * and {@code PixelFormat.BYTE_BGRA_PRE} pixel formats.
+ * A {@code PixelBuffer} can only be drawn using a {@code WritableImage}.
+ * Multiple {@code WritableImage}s and applications can share a {@code PixelBuffer}.
+ * <p>
+ * Pseudo code to create {@code PixelBuffer}:
+ * <pre>{@code
+ *
+ * // Creating a PixelBuffer using PixelFormat.BYTE_BGRA_PRE
+ * ByteBuffer byteBuffer = ByteBuffer.allocateDirect(width * height * 4);
+ * PixelFormat<ByteBuffer> pixelFormat = PixelFormat.getByteBgraPreInstance();
+ * PixelBuffer pixelBuffer = new PixelBuffer(width, height, byteBuffer, pixelFormat);
+ * Image img = new WritableImage(pixelBuffer);
+ *
+ * // Creating a PixelBuffer using PixelFormat.INT_ARGB_PRE
+ * IntBuffer intBuffer = IntBuffer.allocate(width * height);
+ * PixelFormat<IntBuffer> pixelFormat = PixelFormat.getIntArgbPreInstance();
+ * PixelBuffer pixelBuffer = new PixelBuffer(width, height, intBuffer, pixelFormat);
+ * Image img = new WritableImage(pixelBuffer);
+ * }</pre>
+ *
+ * @since 13
+ */
+public class PixelBuffer<T extends Buffer> {
+
+    private final T buffer;
+    private final int width;
+    private final int height;
+    private final PixelFormat pixelFormat;
+    private ArrayList<WeakReference<WritableImage>> imageRefs = new ArrayList<>();
+
+    /**
+     * Construct a {@code PixelBuffer} from the provided {@link java.nio.Buffer} and {@link PixelFormat}.
+     * The constructor does not allocate memory to store the pixel data, application
+     * must allocate sufficient memory required for the dimensions(x, y) and {@code PixelFormat}.
+     * {@code PixelFormat} must be either {@code PixelFormat.INT_ARGB_PRE} or
+     * {@code PixelFormat.BYTE_BGRA_PRE}, otherwise an exception will be thrown.
+     *
+     * @param width       width dimension of the {@code PixelBuffer}.
+     * @param height      height dimension of the {@code PixelBuffer}.
+     * @param buffer      {@code java.nio.Buffer} which stores the pixel data.
+     * @param pixelFormat the {@code PixelFormat} of pixels in the provided {@code buffer}.
+     * @throws IllegalArgumentException if either {@code pixelBuffer} dimension
+     *                                  is negative or zero or if {@code pixelFormat}
+     *                                  is unsupported or if the  {@code buffer} does
+     *                                  not have sufficient memory.
+     * @throws NullPointerException     if {@code buffer} is {@code null}.
+     */
+    public PixelBuffer(int width, int height, T buffer, PixelFormat pixelFormat) {
+        Objects.requireNonNull(buffer, "A valid buffer must be provided.");
+        if (width <= 0 || height <= 0) {
+            throw new IllegalArgumentException("PixelBuffer dimensions must be positive (w,h > 0)");
+        }
+        if (pixelFormat.getType() != PixelFormat.getByteBgraPreInstance().getType() &&
+                pixelFormat.getType() != PixelFormat.getIntArgbPreInstance().getType()) {
```

Comment:
No need to get an instance of a `PixelFormat` for the check. This can be:

```
    if (pixelFormat.getType() != PixelFormat.Type.BYTE_BGRA_PRE &&
            pixelFormat.getType() != PixelFormat.Type.INT_ARGB_PRE) {
```